src/Template/Layout/js/app/components/filter-box.js

```diff
@@ -0,0 +1,186 @@
+ * @prop {String} pageSizesLabel
+ * @prop {String} placeholder
+ * @prop {Boolean} showFilterButtons
+ * @prop {Object} initFilter
+ * @prop {Object} pagination
+ * @prop {String} configPaginateSizes
+ */
+
+import { DEFAULT_PAGINATION } from 'app/mixins/paginated-content';
+
+export default {
+    template:
+    `
+        <nav class="pagination has-text-size-smallest" :class="pagination.count > 4 && 'show-pagination'">
+
+            <div class="pagination-items">
+                <span><: pagination.page_items :> <: objectsLabel :></span>
+            </div>
+
+            <div class="filter-search">
+                <span>
+                    <input type="text" :placeholder="placeholder" v-model="filter" @keyup.enter.prevent.stop="applyFilter()"/>
+                </span>
+
+                <button v-show="showFilterButtons" name="applysearch" @click.prevent="applyFilter()"><: applyFilterLabel :></button>
+                <button v-show="showFilterButtons" name="resetsearch" @click.prevent="resetFilter()"><: resetFilterLabel :></button>
+            </div>
+
+            <div class="page-size">
+                <span><: pageSizesLabel :>:</span>
+            </div>
+
+            <div class="pagination-buttons">
+                <select class="page-size-selector has-background-gray-700 has-border-gray-700 has-text-gray-200 has-text-size-smallest has-font-weight-light" v-model="pageSize">
+                    <option v-for="size in paginateSizes"><: size :></option>
+                </select>
+
+                <button
+                    v-for="i in pagination.page_count" :key="i"
+                    v-if="pagination.page_count > 1"
+                    v-bind:class="pagination.page == i? '' : 'is-dark'"
+                    v-bind:style="pagination.page == i? 'pointer-events: none' : ''"
+                    class="has-text-size-smallest" @click.prevent="changePage(i)"><: i :></button>
+            </div>
+        </nav>
+    `,
+
+    props: {
+        applyFilterLabel: {
+            type: String,
+            default: 'Apply'
+        },
+        resetFilterLabel: {
+            type: String,
+            default: 'Reset',
+        },
+        objectsLabel: {
+            type: String,
+            default: 'Objects',
+        },
+        pageSizesLabel: {
+            type: String,
+            default: 'Size',
+        },
+        placeholder: {
+            type: String,
+            default: 'Search',
+        },
+        showFilterButtons: {
+            type: Boolean,
+            default: true,
+        },
+        initFilter: {
+            type: Object,
+            default: () => {},
+        },
+        pagination: {
+            type: Object,
+            default: () => DEFAULT_PAGINATION,
+        },
+        configPaginateSizes: {
+            type: String,
+            default: '[]',
+        },
+    },
+
+    data() {
+        return {
+            filter: '',                             // Text string filter
```

Comment:
Since it's a new file we could start to set only one space before the inline comment.